### PR TITLE
Hotfix -RPCTimeout

### LINKED ISSUE #936: If RPC is unreachable during running vote command, no error is logged.
#### System information

- **OS & Version**: MacOS
- **Go version:** `go1.17.6 darwin/arm64`
- **razor-go version:** `1.0.0-mainnet stable`

#### Describe the bug
If razor-go is not connected to the internet, and vote command is started, it logs and throws error. But if the user is already voting and in between if there is no internet or RPC is unreachable, nothing is logged. Razor-go becomes stagnant.

#### Expected behaviour
`razor-go` must throw error logs if RPC is unreachable.

#### Actual behaviour
`razor-go` logs nothing.

#### Steps to reproduce the behaviour
1. Start vote command with internet connection.
2. In between turn off the internet.

#### Backtrace
```
{"level":"info","msg":"Block: 321079 Epoch: 1385545 State: -1 Staker ID: 16 Stake: 1172931.560059 sRZR Balance: 113212.859283 Eth Balance: 2.089558","time":"2022-09-08T16:36:34Z"}
{"level":"error","msg":"Error in fetching logs.... Retrying","time":"2022-09-08T16:45:03Z"}
```
For 9 minutes nothing was logged.

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

Added a generic function `InvokeFunctionWithTimeout` to call all the contract and client functions.

Fixes #936

### CODE REVIEW

File: core/constants.go [hunk 1]
```diff
@@ -13,3 +12,6 @@ var StateLength = uint64(EpochLength / NumberOfStates)
 var MaxRetries uint = 8
 var NilHash = common.Hash{0x00}
 var BlockCompletionTimeout = 30
+
+//TODO: Check what should be the timeout
+var RPCTimeout = 10
```
Comment (Yashk767) [inline core/constants.go:17]: Need to discuss. What should be the timeout when RPC is not responding? 
@ashish10677 @SkandaBhat

File: utils/struct-utils.go [hunk 2]
```diff
@@ -51,54 +54,146 @@ func StartRazor(optionsPackageStruct OptionsPackageStruct) Utils {
 	return &UtilsStruct{}
 }
 
+func InvokeFunctionWithTimeout(interfaceName interface{}, methodName string, args ...interface{}) []reflect.Value {
+	var functionCall []reflect.Value
+	var gotFunction = make(chan bool)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(core.RPCTimeout)*time.Second)
```
Comment (Yashk767) [inline utils/struct-utils.go:60]: RPC timeout should be a config parameter

File: utils/struct-utils.go [hunk 3]
```diff
@@ -51,54 +54,146 @@ func StartRazor(optionsPackageStruct OptionsPackageStruct) Utils {
...
+}
+
+func CheckIfAnyError(result []reflect.Value, errorIndexInReturnValues int) error {
+	if result == nil {
+		return errors.New("RPC Timeout error")
+	}
+	if errorIndexInReturnValues == -1 {
+		return nil
+	}
+	returnedError := result[errorIndexInReturnValues].Interface()
```
Comment (Yashk767) [inline utils/struct-utils.go:92]: Don't pass error index to the function, instead implement a `for` loop where you check if the data type is error then handle that error.

**Review by ashish10677 — APPROVED** [review]
lgtm

**Pull request merged**